src/util/server.ml

```diff
@@ -70,17 +70,13 @@ let handle_request (serv: t) (message: Message.either) (id: Id.t) =
 let serve serv =
   serv.input
   |> IO.to_input_channel
-  |> Yojson.Safe.linestream_from_channel
-  |> Stream.iter (fun line ->
-      match line with
-      | `Json json -> (
-          try
-            let message = Message.either_of_yojson json in
-            match message.id with
-            | Some id -> handle_request serv message id
-            | _ -> () (* We just ignore notifications for now. *)
-          with Json.Of_json (s, _) -> prerr_endline s)
-      | `Exn exn -> prerr_endline (Printexc.to_string exn))
+  |> Yojson.Safe.stream_from_channel
+  |> Stream.iter (fun json ->
+      let message = Message.either_of_yojson json in
+      match message.id with
+      | Some id -> handle_request serv message id
+      | _ -> () (* We just ignore notifications for now. *)
+    )
```

Comment:
This conflicts with #591 and I will change this after that's merged, but yes. There's no use for sending invalid JSON to the server, so we might as well go out with a bang in case a broken client does instead of almost silently continuing and appearing to not handle the request.

It actually also changes the input channel parsing such that the JSON objects need not be separated by newlines, because that's a completely artificial limitation. Where one request object ends and another begins is already known by the JSON parser.